### ISSUE TITLE
Package does not load correctly

### ISSUE THREAD

**Issue by fourpastmidnight** [madskristensen/WebCompiler#334]
### Installed product versions
- Visual Studio: 2017 Pro (15.4)
- This extension: 1.11.370

### Description
When starting Visual Studio, and then opening a solution with ASP.NET web projects in it, I get the following dialog at VS Startup, and for each ASP.NET project:

> **Microsoft Visual Studio**  
>  
> The 'WebCompilerInitPackage' package did not load correctly.  
>  
> The problem may have been caused by a configuration change or by  
> the installation of another extension. You can get more information by  
> examining the file '<path to ActivityLog.xml>'.  
>  
> Restarting Visual Studio could help resolve this issue.  
>  
> Continue to show this message?  
>  
> [ Yes ] [ No ]  

I have tried using your "Clear MEF Component Cache" extension to delete the MEF Component Cache to see if that would resolve the issue, but it hasn't. Also, there was an extension that auto-updated today--and sometimes, these things "work themselves out" when other components update, but alas, that has not happened (yet) either.

Lastly, here's the log output from the `ActivityLog.xml` file:

```
LegacySitePackage failed for package [WebCompilerInitPackage]Source:  
'WebCompilerVsix' Description: Could not load file or assembly 'WebCompiler,  Version=1.11.370.0, Culture=neutral, PublicKeyToken=null' or one of its dependencies. 
    The system cannot find the file specified.  
System.IO.FileNotFoundException: Could not load file or assembly 'WebCompiler,  Version=1.11.370.0, Culture=neutral, PublicKeyToken=null' or one of its dependencies. 
    The system cannot find the file specified. 
File name:  'WebCompiler, Version=1.11.370.0, Culture=neutral, PublicKeyToken=null'
    at WebCompilerVsix.WebCompilerInitPackage.Initialize()
    at Microsoft.VisualStudio.Shell.Package.Microsoft.VisualStudio.Shell.Interop.IVsPackage.SetSite(IServiceProvider  sp)
    WRN: Assembly binding logging is turned OFF.
         To enable assembly bind failure  logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog]  (DWORD) to 1.
         Note: There is some performance penalty associated with assembly  bind failure logging.
         To turn this feature off, remove the registry value  [HKLM\Software\Microsoft\Fusion!EnableLog].
--
```

### Steps to recreate
1. Install the latest version of this extension
2. Open Visual Studio. For me, this caused the dialog above to appear.
3. Open a solution with 1 or more ASP.NET projects in it. For me, for each ASP.NET project in the solution, the above dialog was presented.

### Current behavior
YMMV on the steps above, because, as the dialog says, there could be other extensions that are helping to cause this issue. (It could even be as "innocuous" as the order in which the extensions are loaded!)

### Expected behavior
The above dialog is not presented when Visual Studio starts and/or when a solution containing 1 or more ASP.NET web projects are loaded.

### Remarks
This _may_ possibly be caused by a "corrupted" installation of the extension. I'm going to try removing and then adding back this extension. I'll report back if I have any success.

**Comment by fourpastmidnight** [comment]
After uninstalling and then re-installing the extension, the same behavior occurs.

**Comment by jerone** [comment]
Same issue:
```
  <entry>
    <record>1681</record>
    <time>2017/10/14 12:18:27.749</time>
    <type>Information</type>
    <source>VisualStudio</source>
    <description>Begin package load [WebCompilerInitPackage]</description>
    <guid>{6D96A24D-4D89-3A2F-8FF0-8C3DD4D1A789}</guid>
  </entry>
  <entry>
    <record>1682</record>
    <time>2017/10/14 12:18:27.758</time>
    <type>Error</type>
    <source>VisualStudio</source>
    <description>LegacySitePackage failed for package [WebCompilerInitPackage]Source: &apos;WebCompilerVsix&apos; Description: Could not load file or assembly &apos;WebCompiler, Version=1.11.370.0, Culture=neutral, PublicKeyToken=null&apos; or one of its dependencies. The system cannot find the file specified.&#x000D;&#x000A;System.IO.FileNotFoundException: Could not load file or assembly &apos;WebCompiler, Version=1.11.370.0, Culture=neutral, PublicKeyToken=null&apos; or one of its dependencies. The system cannot find the file specified.&#x000D;&#x000A;File name: &apos;WebCompiler, Version=1.11.370.0, Culture=neutral, PublicKeyToken=null&apos;&#x000D;&#x000A;   at WebCompilerVsix.WebCompilerInitPackage.Initialize()&#x000D;&#x000A;   at Microsoft.VisualStudio.Shell.Package.Microsoft.VisualStudio.Shell.Interop.IVsPackage.SetSite(IServiceProvider sp)&#x000D;&#x000A;&#x000D;&#x000A;WRN: Assembly binding logging is turned OFF.&#x000D;&#x000A;To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.&#x000D;&#x000A;Note: There is some performance penalty associated with assembly bind failure logging.&#x000D;&#x000A;To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].&#x000D;&#x000A;</description>
    <guid>{6D96A24D-4D89-3A2F-8FF0-8C3DD4D1A789}</guid>
    <hr>80070002</hr>
    <errorinfo></errorinfo>
  </entry>
  <entry>
    <record>1683</record>
    <time>2017/10/14 12:18:27.772</time>
    <type>Error</type>
    <source>VisualStudio</source>
    <description>SetSite failed for package [WebCompilerInitPackage](null)</description>
    <guid>{6D96A24D-4D89-3A2F-8FF0-8C3DD4D1A789}</guid>
    <hr>80070002</hr>
    <errorinfo></errorinfo>
  </entry>
  <entry>
    <record>1684</record>
    <time>2017/10/14 12:18:27.775</time>
    <type>Error</type>
    <source>VisualStudio</source>
    <description>End package load [WebCompilerInitPackage]</description>
    <guid>{6D96A24D-4D89-3A2F-8FF0-8C3DD4D1A789}</guid>
    <hr>80070002</hr>
    <errorinfo></errorinfo>
  </entry>
```

**Comment by yaorb** [comment]
I has the same issue:
  <description>LegacySitePackage failed for package [WebCompilerPackage]Source: &apos;WebCompilerVsix&apos; Description: Could not load file or assembly“WebCompiler, Version=1.11.370.0, Culture=neutral, PublicKeyToken=null”or one of its dependencies.
The system cannot find the file specified.
System.IO.FileNotFoundException: Could not load file or assembly“WebCompiler, Version=1.11.370.0, Culture=neutral, PublicKeyToken=null”.The system cannot find the file specified.
File Name:“WebCompiler, Version=1.11.370.0, Culture=neutral, PublicKeyToken=null”
   at WebCompilerVsix.CreateConfig..ctor(Package package)
   at WebCompilerVsix.WebCompilerPackage.Initialize() at C:\projects\webcompiler\src\WebCompilerVsix\WebCompilerPackage.cs: row 44
   at Microsoft.VisualStudio.Shell.Package.Microsoft.VisualStudio.Shell.Interop.IVsPackage.SetSite(IServiceProvider sp)
</description>

**Comment by jerone** [comment]
@djanosik Could this be related to your PR https://github.com/madskristensen/WebCompiler/pull/330 (maybe [upgrading .NET from 4.5](https://github.com/madskristensen/WebCompiler/pull/330/files#diff-3835875118d0d911e1ce74f9341de915L12) to 4.6)?

**Comment by djanosik** [comment]
Maybe we need to copy the WebCompiler assembly to output directory when building the extension. At least it's how it works in BundlerMinifier. I didn't know about that. Can you try to fix this?

See: https://github.com/madskristensen/BundlerMinifier/blob/master/src/BundlerMinifierVsix/BundlerMinifierVsix.csproj#L292

**Comment by djanosik** [comment]
Or maybe it's just necessary to update the extension on marketplace? Is the same exception thrown when using nightly build from http://vsixgallery.com/extension/148ffa77-d70a-407f-892b-9ee542346862/?

**Comment by jerone** [comment]
[**@djanosik**](https://github.com/djanosik) commented on [Oct 16, 2017, 10:05 AM GMT+2](https://github.com/madskristensen/WebCompiler/issues/334#issuecomment-336810975 "2017-10-16T08:05:36Z - Replied by Github Reply Comments"):
> Or maybe it's just necessary to update the extension on marketplace? Is the same exception thrown when using nightly build from [http://vsixgallery.com/extension/148ffa77-d70a-407f-892b-9ee542346862/](http://vsixgallery.com/extension/148ffa77-d70a-407f-892b-9ee542346862/)?

The version crashing __is__ the one from VSIX Gallery.

**Comment by djanosik** [comment]
Ok. I will try to update the project file.

**Comment by johancyprich** [comment]
I get the same error, but the plugin still works with compiling SCSS.

**Comment by jerone** [comment]
@djanosik 

This is the content from this extension:
![2017-10-16 19_19_24-eseq1phq lyw](https://user-images.githubusercontent.com/55841/31625630-007c3222-b2a7-11e7-985b-aaf177c6e1b3.png)

Compared to the previous version (different pc):
![image](https://user-images.githubusercontent.com/55841/31625791-a7586890-b2a7-11e7-96f2-5ddbafc2ba48.png)

It looks like I'm missing some files and folders.

**Comment by djanosik** [comment]
@jerone Thank you, it will help.

**Comment by djanosik** [comment]
It should be fixed in #337. @madskristensen Can you merge it please?

**Comment by fourpastmidnight** [comment]
This is fixed for me. Thank you!
